### PR TITLE
[v0.5] Use the proper config when fetching mutating webhook configuration

### PR DESCRIPTION
**Pull request by pmatseykanets**
## Issue: 

https://github.com/rancher/rancher/issues/42611

## Problem

A wrong variable is used when fetching the mutating webhook configuration, that may lead to the following errors on startup:
```
time="2024-07-18T14:01:43Z" level=error msg="Failed to ensure configuration: failed to create mutating configuration: MutatingWebhookConfiguration.admissionregistration.k8s.io \"\" is invalid: metadata.name: Required value: name or generateName is required"
time="2024-07-18T14:01:43Z" level=error msg="error syncing 'cattle-system/cattle-webhook-ca': handler secrets: failed to create mutating configuration: MutatingWebhookConfiguration.admissionregistration.k8s.io \"\" is invalid: metadata.name: Required value: name or generateName is required, requeuing"
```

## Solution

Use the right/intended variable.

## CheckList
  
- [x] Test
  
- [ ] Docs

### CODE REVIEW

**Review by JonCrowther — DISMISSED** [review]

**Review by nflynt — DISMISSED** [review]
I **wondered** why only one of these was getting created when I was doing a docker upgrade reproduction earlier. This is probably why! :+1:

**Comment by ericpromislow** [comment]
I independently came up with the same solution. Was about to figure out how to write an integration test, but given the approvals I assume it's not needed.

Related to issue [#46189](https://github.com/rancher/rancher/issues/46189)

**Comment by nflynt** [comment]
Admittedly I did not think about integration tests, but those are a _really good idea_ since this issue is causing problems in the wild. (It confounds workarounds for the recent docker upgrade kerfuffle, among other things.) We should probably not merge until we have those; I'll gladly re-review when we do.

File: scripts/ci [hunk 1]
```diff
@@ -3,10 +3,10 @@ set -e
 
 cd $(dirname $0)
 
-./build
-./test
```
Comment (pmatseykanets) [inline scripts/ci:7]: Arguably static analysis and other forms of validation should come before building and running tests.

**Review by ericpromislow — APPROVED** [review]
And the update to the test looks fine

File: scripts/ci [hunk 2]
```diff
@@ -3,10 +3,10 @@ set -e
...
 ./validate
 ./validate-ci
+./build
```
Comment (ericpromislow) [inline scripts/ci:8]: This change looks like it's breaking GHA, although I don't see why.

**Review by ericpromislow — CHANGES_REQUESTED** [review]
The amd64 tests are breaking when the validators happen after `build` and `test`. This is probably fixable, but it's not required for this PR.

File: scripts/ci (hunk 2 above)
Comment (pmatseykanets) [reply scripts/ci]: Agreed, reverted. We can look at this outside of this PR.

File: .github/workflows/scripts/integration-test-ci [hunk 3]
```diff
@@ -40,6 +40,9 @@ set -e
 echo "Shutting down core rancher"
 kubectl scale deploy rancher -n cattle-system --replicas=0
 kubectl wait pods -l app=rancher --for=delete -n cattle-system
+# Make sure the webhook recreates configurations on startup
+kubectl delete validatingwebhookconfiguration rancher.cattle.io
+kubectl delete mutatingwebhookconfiguration rancher.cattle.io
```
Comment (pmatseykanets) [inline .github/workflows/scripts/integration-test-ci:45]: This is to make sure we exercise `ensureWebhookConfiguration` when the webhook starts for unit tests.

**Review by ericpromislow — APPROVED** [review]
Looks fine. Why the validators have to happen in the CI after building and testing needs to be left for another PR

**Review by nflynt — APPROVED** [review]
:+1:

**Pull request merged**